src/sqlfluff/rules/references/RF03.py

```diff
@@ -254,40 +261,43 @@ def _validate_one_reference(
                 f"{ref.raw!r} found in single table select which is "
                 "inconsistent with previous references.",
             )
-
+        # Config is consistent, and this reference matches types so far.
         return None
 
-    if single_table_references != this_ref_type:
-        if single_table_references == "unqualified":
-            # If this is qualified we must have a "table", "."" at least
-            fixes = [LintFix.delete(el) for el in ref.segments[:2]] if fixable else None
-            return LintResult(
-                anchor=ref,
-                fixes=fixes,
-                description="{} reference {!r} found in single table "
-                "select.".format(this_ref_type.capitalize(), ref.raw),
-            )
+    # Otherwise check for a specified type of referencing.
+    # If it's the right kind already, just return.
+    if single_table_references == this_ref_type:
+        return None
 
-        fixes = None
-        if fixable:
-            fixes = [
-                LintFix.create_before(
-                    ref.segments[0] if len(ref.segments) else ref,
-                    source=[table_ref_str_source] if table_ref_str_source else None,
-                    edit_segments=[
-                        IdentifierSegment(
-                            raw=table_ref_str,
-                            type="naked_identifier",
-                        ),
-                        SymbolSegment(raw=".", type="symbol"),
-                    ],
-                )
-            ]
+    # If not, it's the wrong type and we should handle it.
+    if single_table_references == "unqualified":
+        # If this is qualified we must have a "table", "."" at least
+        fixes = [LintFix.delete(el) for el in ref.segments[:2]] if fixable else None
         return LintResult(
             anchor=ref,
             fixes=fixes,
             description="{} reference {!r} found in single table "
             "select.".format(this_ref_type.capitalize(), ref.raw),
         )
 
-    return None
+    fixes = None
+    if fixable:
+        fixes = [
+            LintFix.create_before(
+                ref.segments[0] if len(ref.segments) else ref,
+                source=[table_ref_str_source] if table_ref_str_source else None,
+                edit_segments=[
+                    IdentifierSegment(
+                        raw=table_ref_str,
+                        type="naked_identifier",
+                    ),
+                    SymbolSegment(raw=".", type="symbol"),
+                ],
+            )
+        ]
+    return LintResult(
+        anchor=ref,
+        fixes=fixes,
+        description="{} reference {!r} found in single table "
+        "select.".format(this_ref_type.capitalize(), ref.raw),
+    )
```

Comment:
Note for reviewers: This whole section is exactly the same in function as before. All I've done is swapped the condition of the first `if` statement from `!=` to `==`, and by doing that have a first (shorter) `return None` section, and then everything else can be un-nested by one step.